Indexable object is added to ElasticsearchOperationErrorEvent

maijs opened the pull request:
This gives event listeners access to original indexable object if error occurs prior to request wrapper instantiation.

maijs committed:
Indexable object is added to ElasticsearchOperationErrorEvent.

- This gives event listeners access to original indexable object if error occurs
  prior to request wrapper instantiation.

mgalang commented on src/Event/ElasticsearchOperationErrorEvent.php:
```diff
@@ -58,12 +65,14 @@ class ElasticsearchOperationErrorEvent extends Event {
    * @param \Drupal\elasticsearch_helper\ElasticsearchRequestWrapperInterface $request_wrapper
    */
-  public function __construct(\Throwable $error, $operation, ElasticsearchIndexInterface $plugin_instance, ElasticsearchRequestWrapperInterface $request_wrapper = NULL) {
+  public function __construct(\Throwable $error, $operation, ElasticsearchIndexInterface $plugin_instance, $object = NULL, ElasticsearchRequestWrapperInterface $request_wrapper = NULL) {
```
It looks like at least that the Request wrapper is consistently passed to the event object. Perhaps it's best if $object would be the last argument in the constructor, so that we could optionally pass it only when needed instead of setting it as Null in several places.

maijs replied:
~There are more cases where `$request_wrapper` is present whereas `$source` is not, rather than the other way around. For `drop` operation there's never going to be `$source` available, but high probability that `$request_wrapper` is going to be provided.~

I misread the comment which is totally valid.

maijs committed:
Signature for event related methods are changed.

- ElasticsearchIndexBase::dispatchOperationErrorEvent() has $source
  parameter at the end as it's less common than $request_wrapper.
- Constructor of ElasticsearchOperationErrorEvent class has $source
  parameter at the end as it's less common than $request_wrapper.

mgalang reviewed (approved):
Looks good now! Thanks for updates!
